### ISSUE TITLE
Segmentation fault selecting CUDA/VAAPI in video settings

### ISSUE THREAD

**Issue by Gfunk117** [mvasilchuk/yasem#25]
I am currently using gstreamer0.10-ffmpeg for playback and everything works fine but I dont seem to have hardware acceleration with my Nvidia 610 which does support CUDA from what I read on the CUDA Wiki page.  I have all the correct drivers installed I believe but when I select CUDA I get a seg fault and yasem crashes, same with VAAPI in settings...

Is there anyway to get hardware acceleration, vdpau,cuda,vaapi in yasem?  My CPU load on default ffmpeg-gstreamer is around 60 to 70%.

**Comment by mvasilchuk** [comment]
Check what player you're using in yasem. Open Settings->Plugins then you should select either "Qt media player" or "QtAV media player" (if selected both only Qt media player is used) and restart the app.
Qt media player doesn't use OpengGL for rendering so it's slow, but works more stable (it uses system codecs only). QtAV player is less stable and uses it's own codecs, but can render using OpenGL and you may also select a video decoder for it (Setting->Plugins->QtAV->Video settings, restart required; only available when QtAV is selected in plugins).

**Comment by mvasilchuk** [comment]
And about crash. I'll try to fix it soon.

**Comment by Gfunk117** [comment]
Okay when I disable:
qt-mediaplayer=false
libav-mediaplayer=true

I get a segfault on startup and yasem fails to load.  
/home/matt/yasem-x86_64-2015-07-05_12-35-55-2bed7cd# ./yasem
("/opt/Qt5.4.0/5.4/gcc_64/plugins") "."
[LOG  ][21:20:04:845]  Using config directory "/root/.config/yasem/"
[LOG  ][21:20:04:845]  Starting YASEM... Core version: 0.1.0, rev. 8c89768
[LOG  ][21:20:05:294]  Initializing plugin "INI datasource provider"
[STUB ][21:20:05:295] :  "INI datasource provider"
[LOG  ][21:20:05:295]  Initializing plugin "QtAV mediaplayer plugin"
[STUB ][21:20:05:747] :  "QtAV mediaplayer plugin"
[LOG  ][21:20:05:748]  Initializing plugin "Tray icon plugin"
[STUB ][21:20:05:795] :  "Tray icon plugin"
[LOG  ][21:20:05:795]  Initializing plugin "Webkit browser plugin"
[STUB ][21:20:05:796] :  "Webkit browser plugin"
[LOG  ][21:20:05:797]  Initializing plugin "DuneHD JS API implementation"
[STUB ][21:20:05:798] :  "DuneHD JS API implementation"
[LOG  ][21:20:05:799]  Initializing plugin "MAG API implementation"
[STUB ][21:20:05:800] :  "MAG API implementation"
[LOG  ][21:20:05:800]  Initializing plugin "Web GUI"
[STUB ][21:20:05:802] :  "Web GUI"
[LOG  ][21:20:05:802]  Initializing plugin "Desktop GUI plugin"
Segmentation fault (core dumped)

This is on my Nvidia ION with Nvidia Driver 340.76 Ubuntu 15.04 Qt5.4.0

Gonna try the same settings on my Nvidia 610M tomorrow.  I'll post back my results.  I am sure I am missing something.

**Comment by Gfunk117** [comment]
Ill try a git pull and rebuild quick and let you know

**Comment by mvasilchuk** [comment]
Could you, please, run it in debug mode in qtcreator to see where it crashes?

**Comment by Gfunk117** [comment]
Okay I rebuilt yasem and now I do not have any crashes and VAAPI is now working with vdpau-va-driver installed.  CPU usage is around 22% on my test IPTV 720P x264 streams.  

libva info: VA-API version 0.37.0
libva info: va_getDriverName() returns 0
libva info: Trying to open /usr/lib/x86_64-linux-gnu/dri/nvidia_drv_video.so
libva info: Found init function __vaDriverInit_0_35

What driver is used for CUDA acceleration?  I've got libcuda1-340 installed with Nvidia 340 drivers.

Thanks a million for fixing the segfault!  Im loving yasem and I appreciate all the hard work you have put into the development!
